torch_xla/csrc/ops/gather.cpp

```diff
@@ -11,11 +12,24 @@ namespace ir {
 namespace ops {
 namespace {
 
+bool IsSparseGather(const xla::XlaOp& input, const xla::XlaOp& index,
+                    xla::int64 dim) {
+  xla::Shape input_shape = XlaHelpers::ShapeOfXlaOp(input);
+  xla::Shape index_shape = XlaHelpers::ShapeOfXlaOp(index);
+  xla::int64 input_elements = xla::ShapeUtil::ElementsIn(input_shape);
```

Comment:
this seems like it can have its own util function like [tensor -> num_elements]

Reply:
It is not as widely used to have an API IMHO.
The other issue is that getting a Shape of an XLA op is a bit expensive, so I fear folks would use the new API, and then get the Shape in order to access more of its data, effectively materializing it twice.